mule-extensions-api/pom.xml

```diff
@@ -21,6 +21,10 @@
     </properties>
 
     <dependencies>
+        <dependency>
+            <groupId>org.mule.sdk</groupId>
```

Comment:
why? this is very weird and dangerous. Why should this depend on a library which sole purpose is to hide self existence?

Reply:
Many annotations and interfaces of this artifact are getting deprecated in favour of using the ones in mule-sdk-api. When possible, and when it makes sense, we make the interfaces extend from the ones in mule-sdk-api. Should we not do this?